File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.js
function solution(n,a,b)
{
    var answer = 0;
    if (a > b) {
        a, b = b, a;
    }
    while (true) {
        if (a == b) break;
        answer += 1
        if (a % 2) {
            var a_group = Math.floor(a / 2) + 1
        } else {
            var a_group = Math.floor(a / 2)
        }
        if (b % 2) {
            var b_group = Math.floor(b / 2) + 1
        } else {
            var b_group = Math.floor(b / 2)
        }
        a = a_group
        b = b_group
    }


    return answer;
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 0;
        //  1   2   3   4   5  6  7  8  
        // 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16
        
        while (a != b) {
            answer++;
            if (a % 2 == 0) {
                a /= 2;
            } else {
                a = (a+1) / 2;
            }
            
            if (b % 2 == 0) {
                b /= 2;
            } else {
                b = (b+1) / 2;
            }
        }
        return answer;
    }
}